compat/os/mingw.h

```diff
@@ -32,5 +32,17 @@
 #include <windows.h>
 #endif
 
+#if HAVE_IO_H
+#include <io.h>
+#endif
```

Comment:
Adding, but:
<io.h> it is also included, without tests, in case of _SQUID_WINDOWS_,
in compat/os/mswindows.h, lib/dirent.c, and src/auth/basic/RADIUS/basic_radius_auth.cc